internal/clients/auth.go

```diff
@@ -46,6 +47,16 @@ func NewResourceManagerAccount(ctx context.Context, config auth.Credentials, sub
 		return nil, fmt.Errorf("parsing claims from access token: %+v", err)
 	}
 
+	// Log the claims for debugging
+	claimsJson, err := json.Marshal(claims)
+	if err != nil {
+		log.Printf("[DEBUG] AzureRM Provider could not marshal access token claims for log outout")
+	} else if claimsJson == nil {
+		log.Printf("[DEBUG] AzureRM Provider marshaled access token claims was nil")
+	} else {
+		log.Printf("[DEBUG] AzureRM Provider access token claims: %s", claimsJson)
+	}
```

Comment:
I think we'd get the same effect from:

```suggestion
	// Log the claims for debugging (NOTE: these aren't considered sensitive)
	claimsJson, err := json.Marshal(claims)
	if err != nil {
		log.Printf("[DEBUG] AzureRM Provider could not marshal access token claims for log outout")
	}
	
	log.Printf("[DEBUG] AzureRM Provider access token claims: %q", string(claimsJson))
```